active/003-seq-feature.md

```diff
@@ -0,0 +1,98 @@
+- `length` : int.  The length of the genome.  This is required when swapping coordinates
+- This performs a reverse complement on all the coordinates with in IntervalTree.
+
+```python
+   interval_metadata = IntervalMetadata()
+   interval_metadata.add(Feature(gene='sagB', location=0), (3, 5))
+   iv = interval_metadata.reverse_complement(length=10)
+```
+
+`update(old_feature, new_feature)`
+- `old_feature` : `skbio.Feature`.  feature to be replaced
+- `new_feature` : `skbio.Feature`.  updated feature to replace the original feature
+- Since `skbio.Feature` objects are immutable, if individual features in the `IntervalMetadata` are to be updated, the need to be completely replaced.
+
+```python
+    interval_metadata = IntervalMetadata()
+        interval_metadata.add(Feature(gene='sagA', location=0), 1, (4, 7))
+        interval_metadata.add(Feature(gene='sagB', location=0), (3, 5))
+        interval_metadata.update(Feature(gene='sagB', location=0),
+                                 Feature(gene='sagB', location=1))
+```
+
+
+`add(feature, *intervals)`
+- `feature` : `skbio.Feature`. new feature object add into the `IntervalMetadata` object
+- `*intervals` : a list of tuples corresponding to intervals where the feature object exists.
+- This allows for a single feature (include those that have non-continugous features) to be added into the `IntervalMetadata` object.
+
+```python
+   interval_metadata = IntervalMetadata()
+   interval_metadata.add(Feature(gene='sagA', location=0), 1, (4, 7))
+   interval_metadata.add(Feature(gene='sagB', location=0), (3, 5))
+```
+
+`query(*args, **kwargs)`
+- `*args` : a list of tuples to search for features
+- `**kwargs` : keyword arguments to query features by their attributes.
+- This allows for features to be searched by both intervals and keywords (i.e. CDS vs exon, ...)
+
+```python
+   interval_metadata = IntervalMetadata(features={
+                                        Feature(gene='sagA', location=0): [(0, 2), (4, 7)],
+                                        Feature(gene='sagB', location=0): [(3, 5)]})
+ 
+   feats = interval_metadata.query((1, 2))
+   feats = interval_metadata.query(gene='sagB')
+```
+
+This is the current implementation in micronota and fits all of our use-cases at the moment.  We don't feel that IntervalIndex objects as suggested in the recent pandas poll request can easily handle all of our use cases (i.e. non-contiguous intervals isn't straightforward).
+
+##Feedback
+- The `update` method may not be the most straightforward for users.  The reason why we opted for this design was because the `skbio.Feature` object can only be indexed uniquely by hashing the entire feature.  Suggestions on how to improve this will be welcome.
+- The `query` method does a linear lookup when searching for attributes over features.  This problem can blow up (in terms of development) if we want faster querying.  Which is why we opted for slow, linear search.  Again suggestions here are welcome.
```

Comment:
These concerns suggest indexing, but given the schemaless nature of the `Feature`s, doing indexing here may be worth discussing later.